proposal/package-importer.md

```diff
@@ -0,0 +1,191 @@
+
+A potential source of confusion is in instances where a dependency defines a
+default or main export that is not in the root folder. In these cases, Sass will
+look for index and partial files in the same folder as the default or main
+export, and not in the package root.
+
+The `pkg` scheme will not be supported in the browser version of Dart Sass. To
+support a similar functionality, a user would need to ensure that files are
+served, and the loader would need to fetch the URL. In order to follow the same
+algorithm for [resolving a file: URL], we would need to make many fetches. If we
+instead require the browser version to have a fully resolved URL, we negate many
+of this spec's benefits.
+
+[resolving a file: URL]: ../spec/modules.md#resolving-a-file-url
+
+## Semantics
+
+This proposal defines a new step in the [Loading a Source File] in modules that
+implementations must include.
+
+New step for [Loading a Source File], after "if argument is a relative URL"
+step:
+
+- If `argument` is a valid URL with scheme `pkg`:
+  - Let `resolved` be the result of resolving a pkg url
+  - If `resolved` is not null, let `argument` equal `resolved`, and continue.
+    Otherwise, return null.
+
+[Loading a Source File]: ../spec/modules.md#loading-a-source-file
+
+### Resolving a `pkg`: URL
+
+This algorithm takes a URL, `url`, whose `scheme` must be `pkg` and returns
+either another URL of a file or directory on disk, or null. At this point, the
+URL is not guaranteed to resolve to a file or disk.
+
+- Let `resolved` be the result of the implementation's dependency resolution
+  algorithm.
+- If resolution fails:
+  - Let `dependencyDefault` be the result of the implementation's dependency
+    resolution algorithm for the dependency name.
+  - Let `urlSubpath` be the path segments of `url` that are not part of the
+    dependency name.
+  - Let `resolved` be `dependencyDefault` appended with `urlSubpath`.
+- If `resolved` is unknown, meaning neither the full path nor the dependency
+  name were resolved, return `null`.
+- If `resolved` is a directory or a file with extension of `scss`, `sass`, or
+  `css`, return it. Otherwise:
+- If `resolved` is a file with a name that is different than the name of `url`,
+  return the file's containing directory.
+- Return null.
+
+Note: The fifth step, returning the containing directory, is required to support
+instances where a dependency's default export is not a `sass`, `scss` or `css`
+file.
+
+[Loading a Module]: ../spec/modules.md#loading-a-source-file
+
+## Deprecation Process
+
+The `package` url scheme supported in Dart is not part of the Sass spec, but is
+supported by the `dart-sass` implementation when running in Dart. This should be
+deprecated in favor of moving authors to the new `pkg` url scheme. Usage of the
+`package` syntax will result in a deprecation message, and a future major
+version of Sass should remove support.
+
+## Examples
+
+### Node
```

Comment:
Unfortunately, we will need a more formal specification for the Node importer. The JS API will need to allow users to opt into or out of it (I'm not sure which is the better default), and if it's touched by the JS API its behavior needs to be fully specified. We'll probably also want to expose it to the embedded protocol, since Node.js is so pervasive in the frontend world and it's likely that even other embedded hosts will sometimes want to load dependencies from `node_modules`.

Reply:
Added in [14b5df6](https://github.com/oddbird/sass/pull/3/commits/14b5df60a4becb03470e4b6a752b89eede2eee1d), although I deferred to the existing Modules spec where possible.